### PR TITLE
circulation: update ILL request JSON schema

### PR DESCRIPTION
**Pull request by zannkukai**
* Allows to manage manually the loan status.
* Adds notes (public/staff) to an ILL request.
* Configures request resource to allow faceting.
* Fixes search serializer configuration to include created and updated
  date into search results.
* Fixes object serializer with `resolve` argument. 'created' and
  'updated' keys were null, date are now correct.

Closes rero/rero-ils#1320

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Comment by ManaDeweerdt** [comment]
Cannot see the 'add' button in the ILL requests result if I'm logged in as a librarian. (ok for the system librarian)

**Comment by ManaDeweerdt** [comment]
When I add a ILL request for a patron, if I don't fill the patron field, it spins forever.

**Comment by ManaDeweerdt** [comment]
I can choose patron who has only 'librarian' as role. Ex : Elena Rodriguez. --> The requester field is empty. 
image

**Comment by pronguen** [comment]
- [ ] The field label "Source" on the request detailed view should be named "Published in"
- [ ] The field "Trouvé dans" should be placed in the first section of the detailed view (this is actually also the case in the public interface)
- [ ] The field "Trouvé dans" should be placed in the first section of the editor (see next comment)
- [ ] Is is possible in the editor to have subsections, so that the librarian see quickly where to find the loan status and request statuts? I imagine the same sections than in the detailed view: Information about the document, Information about the request.
- [ ] A librarian (not system librarian) should be able to edit only the request of her/his library (to be confirmed by @iGormilhit )
- [ ] When a librarian adds a request, it spins forever if a required field is not entered (patron, pick-up library).
- [ ] To ease the translations, I would put the same string "no data" everywhere, and would not use "no source". I am wondering if it is really relevant to display this lack of information and would prefer a shorter detailed view.
image
- [ ] Cannot see the 'add' button in the ILL requests result if I'm logged in as a librarian (Mana's comment)

File: rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json [hunk 1]
```diff
@@ -188,26 +252,94 @@
...
+          "content"
+        ],
+        "required": [
+          "type",
+          "content"
+        ],
+        "properties": {
+          "type": {
+            "type": "string",
+            "title": "Type",
```
Comment (iGormilhit) [inline rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json:294]: Properties order (in the schema, not for the editor):
```suggestion
            "title": "Type",
            "type": "string",
```

File: rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json [hunk 2]
```diff
@@ -188,26 +252,94 @@
...
+                {
+                  "label": "staff_note",
+                  "value": "staff_note"
+                }
+              ]
+            }
+          },
+          "content": {
+            "type": "string",
+            "title": "Content",
```
Comment (iGormilhit) [inline rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json:316]: Properties order: title and type.

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
circulation: update ILL request JSON schema

* Links an ILL request to a loan.
* Adds notes (public/staff) to an ILL request.
* Configures request resource to allow faceting.
* Fixes search serializer configuration to include created and updated
  date into search results.
* Fixes object serializer with `resolve` argument. 'created' and
  'updated' keys were null, date are now correct.
```
Is there actually a link between an ILL request and a loan? Or just a way to update the `loan_status` field manually (as I expected)?

File: rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json (hunk 1 above)
Comment (zannkukai) [reply rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json:294]: done

**Comment by BadrAly** [comment]
I do not think `Allows to link an ill_request to a loan` is valid. as I see it is the librarian to manually enter the loan status

**Review by BadrAly — APPROVED** [review]

**Review by BadrAly — CHANGES_REQUESTED** [review]
Test is failing

**Review by lauren-d — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Review by BadrAly — APPROVED** [review]

**Pull request merged**